src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -1403,6 +1403,7 @@ impl GraphEditorModel {
     /// implementation.
     fn remove_node(&self, node_id:impl Into<NodeId>) {
         let node_id = node_id.into();
+        self.frp.source.on_visualization_select.emit(Switch::Off(node_id));
```

Comment:
Emiting frp events from model functions is rather a bad pattern, as it is really hard to understand where the events come from. Instead, check where this model function is called in the FRP network and handle it directly there.

This should be handled next to the lines:

```
output.source.on_visualization_select <+ selected.constant(Switch::On(node_id));
output.source.on_visualization_select <+ deselected.constant(Switch::Off(node_id));
```

so it would be easy to track when visualization is deselectected when reading the code (all deselecting will be coded in one place then)